src/main.tsx

```diff
@@ -356,25 +389,188 @@ export default class AnnotatorPlugin extends Plugin implements IHasAnnotatorSett
     private addMarkdownPostProcessor() {
         const markdownPostProcessor = async (el: HTMLElement, ctx: MarkdownPostProcessorContext) => {
             for (const link of el.getElementsByClassName('internal-link') as HTMLCollectionOf<HTMLAnchorElement>) {
-                const parsedLink = parseLinktext(link.href);
-                const annotationid = parsedLink.subpath.startsWith('#^') ? parsedLink.subpath.substr(2) : null;
+                const parsedLink = parseLinktext(link.getAttribute("href"));
+                const annotationid = parsedLink.subpath.startsWith('#^') ? parsedLink.subpath.substring(2) : null;
                 const file: TFile | null = this.app.metadataCache.getFirstLinkpathDest(parsedLink.path, ctx.sourcePath);
 
                 if (this.isAnnotationFile(file)) {
-                    link.addEventListener('click', ev => {
-                        ev.preventDefault();
-                        ev.stopPropagation();
-                        ev.stopImmediatePropagation();
-                        const inNewPane = ev.metaKey || ev.ctrlKey || ev.button == 1;
-                        this.openAnnotationTarget(file, inNewPane, annotationid);
-                    });
+                    this.addClickListener(link, annotationid, file);
                 }
             }
         };
 
         this.registerMarkdownPostProcessor(markdownPostProcessor);
     }
 
+    addClickListener(element: HTMLAnchorElement, annotationid: string, file: TFile) {
+        const childs = element.children;
+        if (childs && childs.length == 1) {
+            element.addEventListener('click', ev => {
+                this.log(annotationid);
+                ev.preventDefault();
+                ev.stopPropagation();
+                ev.stopImmediatePropagation();
+                const inNewPane = ev.metaKey || ev.ctrlKey || ev.button == 1;
+                this.openAnnotationTarget(file, inNewPane, annotationid);
+            });
+        }
+    }
+
+    linkOnFocus(element: Element) {
+        let count = 0;
+        const linkText = element.textContent;
+        while(element.className && element.className.indexOf('cm-formatting-link cm-formatting-link-end') == -1) {
+            if (this.targetClassNameSet.has(element.className)) count++;
+            element = element.nextElementSibling;
+        }
+        this.tmpLinkInfos.push({linkText: linkText, count: count});
+    }
+
+    linkOnBlur(node: Element, rawLinkText: string, filePath: string, className: string) {
```

Comment:
When the link loses the cursor

![0004](https://user-images.githubusercontent.com/45873697/201304476-13b7ebf5-1905-4605-b88f-8280c0af7b93.gif)